src/main/java/org/cactoos/text/PrefixOf.java

```diff
@@ -40,25 +40,34 @@
 
     /**
      * Ctor.
-     * @param text Text representing the text value
-     * @param boundary String to which text will be split
+     * @param text CharSequence representing the text value
+     * @param boundary CharSequence to which text will be split
      */
-    public PrefixOf(final String text, final String boundary) {
+    public PrefixOf(final CharSequence text, final CharSequence boundary) {
         this(new TextOf(text), boundary);
     }
 
+    /**
+     * Ctor.
+     * @param text Text representing the text value
+     * @param boundary CharSequence to which text will be split
+     */
+    public PrefixOf(final Text text, final CharSequence boundary) {
+        this(text, new TextOf(boundary));
+    }
+
     /**
      * Ctor.
      * @param text Text representing the text value
      * @param boundary String to which text will be split
      */
-    public PrefixOf(final Text text, final String boundary) {
+    public PrefixOf(final Text text, final Text boundary) {
```

Comment:
@andreoss I would prefer that we don't change the API: here the `text` object is the one wrapped while `boundary` is just data, so let's keep it as `CharSequence` and not `Text` (because as you can see below, `asString()` is called multiple times and it could create surprising behaviour if `asString()` does not always return the same thing: that's why the `text` object itself is wrapped in Sticky inside the `Ternary` btw).